packages/web/src/components/ProjectRegistrationModal/RegistrationForm.tsx

```diff
@@ -0,0 +1,149 @@
+import { useFormik } from 'formik';
+import React, { useEffect, useState } from 'react';
+import * as Yup from 'yup';
+import {
+  Input,
+  FormControl,
+  VStack,
+  FormHelperText,
+  FormLabel,
+  Alert,
+  AlertDescription,
+  AlertIcon,
+  AlertTitle,
+  CloseButton,
+  Button,
+} from '@chakra-ui/react';
+
+interface ProjectProps {
+  name?: string;
+  description?: string;
+  tableNumber?: number;
+}
+
+export const RegistrationForm: React.FC = () => {
+  const [serverError, setServerError] = useState('');
+  const [project, setProject] = useState<ProjectProps>({});
+  const [validateWhileTyping, setValidateWhileTyping] = React.useState(false);
+
+  useEffect(() => {
+    const fetchProjects = async () => {
+      // todo default endpoint that user req.user to find the user's project/team
+      const res = await fetch('/api/project/');
+
+      try {
+        if (!res.ok) {
+          throw new Error();
+        }
+
+        const data = await res.json();
+        setProject(data);
+      } catch (err) {
+        // TODO print/log error message to show we were unable to pull their project data
+        setProject({});
+      }
+    };
+
+    void fetchProjects();
+  }, []);
+
+  const RegistrationSchema = Yup.object().shape({
+    name: Yup.string().required('Name is required'),
+    description: Yup.string(),
+    tableNumber: Yup.number(),
+  });
+
+  const formik = useFormik({
+    initialValues: {
+      name: project.name,
+      description: project.description,
+      tableNumber: project.tableNumber,
+    },
+    validationSchema: RegistrationSchema,
+    validateOnBlur: validateWhileTyping,
+    validateOnChange: validateWhileTyping,
+    async onSubmit(values) {
+      const res = await fetch('/api/project/register', {
+        method: 'POST',
+        body: JSON.stringify({
+          inputConfig: values,
+        }),
+        headers: {
+          'Content-Type': 'application/json',
+        },
+      });
+      if (!res.ok) {
+        setServerError('Something went wrong, please try again later...');
+      }
+    },
+  });
+
+  useEffect(() => {
+    if (formik.submitCount && !validateWhileTyping) {
+      setValidateWhileTyping(true);
+    }
+  }, [formik.submitCount, validateWhileTyping]);
+
+  return (
+    <form onSubmit={formik.handleSubmit}>
+      <VStack alignItems="stretch">
+        <FormControl id="name">
+          <FormLabel>Name</FormLabel>
+          <Input
+            type="text"
+            placeholder="Your project name"
+            value={formik.values.name ?? project.name}
+            isInvalid={!!formik.errors.name}
+            onChange={formik.handleChange}
+            onBlur={formik.handleBlur}
+            disabled={formik.isSubmitting}
+          />
+          <FormHelperText color="red.500">{formik.errors.name}&nbsp;</FormHelperText>
+        </FormControl>
+        <FormControl id="description">
+          <FormLabel>Description</FormLabel>
+          <Input
```

Comment:
See if a `TextArea` will work here - if not, let's chat and see if it's worth the effort